### ISSUE TITLE
Why the softmax with product?

### ISSUE THREAD

**Issue by ghosthamlet** [guillaumegenthial/tf_ner#18]
Why the local softmax formula in your blog: https://guillaumegenthial.github.io/sequence-tagging-with-tensorflow.html used product, but  the code just have normal softmax:
 losses = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=scores, labels=labels)，

And this sentence aslo confused me: "Eventually, the probability ℙ(y) of a sequence of tag y is the product pt"，when softmax will need the product?

**Comment by guillaumegenthial** [comment]
The function `sparse_softmax_cross_entropy_with_logits` is an efficient way of computing the softmax (see the doc). It should be equivalent to what is written on the post.

The sentence is a application of a basic probability theorem : the probability of independent events is the product of their probabilities. In the softmax setting, we assume that every tag is assigned independently of the others. Each probability is computed via the softmax. Then the probability of the whole sentence is the product of those probabilities.
